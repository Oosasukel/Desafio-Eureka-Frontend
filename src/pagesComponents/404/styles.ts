import styled from 'styled-components';

export const Container = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  overflow: auto;
`;

export const AnimationWrapper = styled.div`
  > * {
    min-width: 16rem;
    width: 32vw !important;
    height: auto !important;
  }
`;
