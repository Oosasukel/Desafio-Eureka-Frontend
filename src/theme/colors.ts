export const colors = {
  background: '#FCFAF9',
  primary: '#333333',
  secondary: '#989796',
  danger: '#FF8C8C',
} as const;
