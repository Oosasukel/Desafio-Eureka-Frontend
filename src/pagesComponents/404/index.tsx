import notFoundAnimation from 'assets/lotties/not-found-ghost.json';
import Lottie from 'react-lottie';
import * as S from './styles';

export const PageComponent404 = () => {
  return (
    <S.Container>
      <S.AnimationWrapper>
        <Lottie options={{ animationData: notFoundAnimation }} />
      </S.AnimationWrapper>
    </S.Container>
  );
};
