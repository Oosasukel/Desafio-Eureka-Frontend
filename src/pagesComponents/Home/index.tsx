import { AddressCard } from './components/AddressCard';
import { Search } from './components/Search';
import { HomeProvider } from './Provider/Provider';
import * as S from './styles';

export const Home = () => {
  return (
    <HomeProvider>
      <S.Container>
        <S.Title>Encontre um CEP</S.Title>
        <S.Subtitle>
          Digite um CEP abaixo e obtenha todas as informações sobre ele.
        </S.Subtitle>

        <Search />
        <AddressCard />
      </S.Container>
    </HomeProvider>
  );
};
