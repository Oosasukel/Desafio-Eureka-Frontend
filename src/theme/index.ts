import { colors } from './colors';
import { fontWeights } from './fontWeights';

export const theme = {
  colors,
  fontWeights,
};
