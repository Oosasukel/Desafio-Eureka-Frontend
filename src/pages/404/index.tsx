import { PageComponent404 } from 'pagesComponents/404';

const Page404 = () => <PageComponent404 />;

export default Page404;
