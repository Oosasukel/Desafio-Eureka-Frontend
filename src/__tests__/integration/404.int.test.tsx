import { render } from '@testing-library/react';
import { PageComponent404 } from 'pagesComponents/404';
import { ThemeProvider } from 'styled-components';
import { theme } from 'theme';

describe('Page Home', () => {
  it('should match snapshot', async () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});

const renderComponent = () =>
  render(
    <ThemeProvider theme={theme}>
      <PageComponent404 />
    </ThemeProvider>
  );
