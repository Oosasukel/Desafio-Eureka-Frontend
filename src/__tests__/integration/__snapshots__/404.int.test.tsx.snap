// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Home should match snapshot 1`] = `
<div>
  <div
    class="styles__Container-sc-1pzvp8o-0 fVDrnI"
  >
    <div
      class="styles__AnimationWrapper-sc-1pzvp8o-1 HShWS"
    >
      <span>
        Lottie Animation
      </span>
    </div>
  </div>
</div>
`;
